esmvaltool/preprocessor/_derive/__init__.py

```diff
@@ -55,8 +52,8 @@ def get_required(short_name):
         and occasionally mip or fx_files.
 
     """
-    DerivedVariable = ALL_DERIVED_VARIABLES[short_name]
-    variables = deepcopy(DerivedVariable().required)
+    derived_variable = ALL_DERIVED_VARIABLES[short_name]
```

Comment:
Change this back to what it was, this is actually the DerivedVariable class associated with the variable we want to derive. Codacy is just wrong here.